test/PandoraInterface.cxx

```diff
@@ -336,14 +368,464 @@ void ProcessEvents(const Parameters &parameters, const Pandora *const pPrimaryPa
+        if (ok)
+        {
+            point2 = ray.getPoint(t1);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+    else if (!inStart && inStop)
+    {
+        //std::cout << "Case 4: End inside boundary" << std::endl;
+        point2 = stop;
+        const bool ok = grid.intersect(ray, t0, t1);
+        if (ok)
+        {
+            point1 = ray.getPoint(t0);
+        }
+        else
+        {
+            std::cout << "Voxel ray does not intersect boundary" << std::endl;
+            return currentVoxelList;
+        }
+    }
+
+    // Now create voxels between point1 and point2.
+    // Ray direction will be the same, but update starting point
+    ray.updateOrigin(point1);
+
+    // Small position shift to ensure we move along the ray path
+    const double epsilon(1e-3);
+    bool shuffle(true);
+
+    // Keep track of total voxel path length and energy (so far)
+    double totalPath(0.0), ETot(0.0);
+    int loop(0);
+
+    while (shuffle)
+    {
+
+        // Get point along path to define voxel bin (bottom corner)
+        pandora::CartesianVector voxelPoint = ray.getPoint(epsilon);
+
+        // Grid 3d bin containing this point; 4th element is the total bin number
+        std::array<long, 4> gridBins = grid.getBinIndices(voxelPoint);
+        long voxelID = gridBins[3];
+
+        long xBin = gridBins[0];
+        long yBin = gridBins[1];
+        long zBin = gridBins[2];
+
+        // Voxel bottom and top corners
+        const pandora::CartesianVector voxBot = grid.getPoint(xBin, yBin, zBin);
+        const pandora::CartesianVector voxTop = grid.getPoint(xBin + 1, yBin + 1, zBin + 1);
+
+        // Voxel box
+        const LArBox vBox(voxBot, voxTop);
+        // Get ray intersections with this box: t0 and t1 are set as the start
+        // and end intersection pathlengths relative to the current ray point
+        bool result = vBox.intersect(ray, t0, t1);
+
+        if (!result)
+        {
+            shuffle = false;
+        }
+
+        // Voxel extent = intersection path difference
+        double dL(t1 - t0);
+        // For the first path length, use the distance from the
+        // starting ray point to the 2nd intersection t1
+        if (loop == 0)
+        {
+            dL = t1;
+        }
+
+        // Stop processing if we are not moving the path along
+        if (dL < epsilon)
+        {
+            shuffle = false;
+        }
+
+        totalPath += dL;
+
+        // Stop adding voxels if we have enough
+        if (totalPath > hitLength)
+        {
+            shuffle = false;
+            // Adjust final path according to hit segment total length
+            dL = hitLength - totalPath + dL;
+        }
+
+        // Voxel energy using path length fraction w.r.t hit length
+        const double voxelEnergy = g4HitEnergy * dL / hitLength;
```

Comment:
hitLength zero checked anywhere? double vs. float.

Reply:
Checks are made on hit length and hit energy to be > 0. Changed to float.